src/streaming/multiplex/advanced.rs

```diff
@@ -304,7 +304,9 @@ impl<T> Multiplex<T> where T: Dispatch {
         while self.run {
             // TODO: Only read frames if there is available space in the frame
             // buffer
-            if let Async::Ready(frame) = try!(self.dispatch.get_mut().inner.transport().poll()) {
+            if self.frame_buf.full() {
```

Comment:
It's been a while since I dealt with this code. When the frame_buf has capacity again, something has to notify the task so that it starts running again. Could you explain how this happens?